currency.py

```diff
@@ -0,0 +1,83 @@
+import urllib.request
+import sys
+import time
+
+
+def leap_year_check(year):
+    if year % 4 == 0 and (year % 400 == 0 or year % 100 != 0):
+        return True
+    else:
+        return False
+
+
+def days_in_month(month, year):
```

Comment:
You've hit the nail on the date parsing algorithms: they can be really complicated. I see that you've tried to detect the number of days in February depending on the year, etc.

However, Python already contains that stuff in the standard library. Here is how to do that:

```python
from datetime import datetime

print(datetime.strptime("10/01/2020", "%d/%m/%Y"))
```

So this function will parse an arbitrary date string for you depending on the date format you've passed as the second argument. Here is the complete list of available date format placeholders: https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes

In our current situation, you need only three of them: `%d` (days in format 01-31), `%m` (months in format 1-12), and `%Y` which represents the year with four numbers.

This function returns the `datetime` object, which contains information about the month, day, year, hours, seconds, etc. as separate variables. For example, here is how you can get the year from it:

```python
from datetime import datetime

dt = datetime.strptime("2020/01/01", "%Y/%m/%d")
print(dt.year)
```
Once you've got the `datetime` object, you can convert it back to a string representation using the specific date format again. For example:

```python
from datetime import datetime

dt = datetime.strptime("2020/01/01", "%Y/%m/%d")
print(dt.strftime("%B %d, %Y")) # will give you 'January 01, 2020'
```

To summarize, there are two functions:
1. `datetime.strptime()` which can parse your date
2. `datetime.strftime()` which can convert your datetime object to a string in accordance with specific date format

Try to use them instead of your own code